[pylint] Fix ``consider-using-sys-exit``, ``use-yield-from``, ``implicit-str-concat`` and ``pointless-x-statement``

Pierre-Sassoulas opened the pull request:
Follow-up to #12172

bluetech reviewed (commented):
IMO, the `pointless-x-statement` changes don't really improve things, I'd leave it out (but I don't feel strongly about this).

I think there is some confusion in the commit subject `[pylint 'implicit-str-concat'] Use zip when iterating on two iterators`.

Other than that LGTM, thanks!

Pierre-Sassoulas commented:
> IMO, the pointless-x-statement changes don't really improve things, I'd leave it out (but I don't feel strongly about this).

Removed (the benefit would be to be able to see new violations without having low values warning, I agree that the diff is not much better). Either ruff did not catch them or they filtered this when in a pytest.raises / unitest.assertRaises, which make sense.

> I think there is some confusion in the commit subject [pylint 'implicit-str-concat'] Use zip when iterating on two iterators.

Yeah, nice catch, probably reused an old message without updating properly, fixed.
